File: docs/examples/python/heightheight.py

# <snippet>

import GooseEYE as eye
import numpy    as np

x = np.linspace(0, 2.0 * np.pi, 1000)
y1 = np.sin(x)
y2 = np.sin(2. * x)

hh1 = eye.heightheight(roi=[200], f=y1, periodic=True)
hh2 = eye.heightheight(roi=[200], f=y2, periodic=True)
dx = eye.distance(roi=[200], dim=0)

# </snippet>

# plot
# ----

import matplotlib.pyplot as plt

try   : plt.style.use(['goose','goose-latex'])
except: pass

fig, axes = plt.subplots(ncols=2)

ax = axes[0]

ax.plot(x, y1)
ax.plot(x, y2)

ax = axes[1]

ax.plot(dx, hh1)
ax.plot(dx, hh2)

plt.show()

File: docs/examples/heightheight.cpp
#include <GooseEYE/GooseEYE.h>
#include <highfive/H5Easy.hpp>

#define MYASSERT(expr) MYASSERT_IMPL(expr, __FILE__, __LINE__)
#define MYASSERT_IMPL(expr, file, line) \
    if (!(expr)) { \
        throw std::runtime_error( \
            std::string(file) + ':' + std::to_string(line) + \
            ": assertion failed (" #expr ") \n\t"); \
    }

int main()
{
    double L = 2.0 * M_PI;
    size_t N = 1000;
    double h = L / (double)N;
    xt::xarray<double> x = xt::linspace<double>(0, L, N);
    xt::xarray<double> y1 = xt::sin(x);
    xt::xarray<double> y2 = xt::sin(2. * x);

    std::vector<size_t> roi = {200};
    std::vector<double> hv = {h};
    xt::xarray<double> hh1 = GooseEYE::heightheight(roi, y1, true);
    xt::xarray<double> hh2 = GooseEYE::heightheight(roi, y2, true);
    xt::xarray<double> dx = GooseEYE::distance(roi, hv, 0);

    // check against previous versions
    H5Easy::File data("heightheight.h5", H5Easy::File::ReadOnly);
    MYASSERT(xt::allclose(y1, H5Easy::load<decltype(y1)>(data, "y1")));
    MYASSERT(xt::allclose(y2, H5Easy::load<decltype(y2)>(data, "y2")));
    MYASSERT(xt::allclose(hh1, H5Easy::load<decltype(hh1)>(data, "hh1")));
    MYASSERT(xt::allclose(hh2, H5Easy::load<decltype(hh2)>(data, "hh2")));
    MYASSERT(xt::allclose(dx, H5Easy::load<decltype(dx)>(data, "dx")));

    return 0;
}
